### ISSUE TITLE
Rename "bassloopes" folder to "bassloops".

### ISSUE THREAD

**Issue by sunnystormy** [LMMS/lmms#2617]
Fix spelling error, please. :)

**Comment by tresf** [comment]
For the person who tackles this, here's an example where we had a misspelling and used regex to replace the value in the project file as part of the `DataFile::upgrade()` process.

https://github.com/LMMS/lmms/blob/master/src/core/DataFile.cpp#L722:l723

**Comment by T-boneG** [comment]
Hey there!  I would be happy to take this on as an inspiring programmer hoping to get involved!  What is the standard etiquette on assigning bugs?

**Comment by tresf** [comment]
> What is the standard etiquette on assigning bugs?

Thanks!  P.S. "I got it" is perfectly valid for minor items like this, so your etiquette is just fine.  P.S. Welcome!

**Comment by tresf** [comment]
@T-boneG do you need any assistance getting started on this one?

**Comment by T-boneG** [comment]
Hi Tres!

Thanks for reaching out!  I am getting way more comfortable with the code
and in general using Qt creator.  I think I found a solution:

		{

			QDomNodeList list = elementsByTagName( "audiofileprocessor" );

			for( int i = 0; !list.item( i ).isNull(); ++i )

			{

				QDomElement el = list.item( i ).toElement();

				QString s = el.attribute( "src" );

				s.replace( QRegExp("bassloopes"), "bassloops" );

				el.setAttribute( "src", s );

			}

		}



Which I am calling from within: DataFile::loadData( const QByteArray & _data
, const QString & _sourceFile )

I am wondering about a few things though:

1)  Despite the very low likelihood of someone making their own file or
folder named "bassloopes", it still seems unsafe to change the path without
knowing if this is an LMMS default or a user's personal file or folder.  Is
this a concern?  What's a good way to go about preventing it?

2) Should I write my own upgrade for this?

-Tyler

**Comment by eagles051387** [comment]
Good evening ,

Just out of curiosity wouldn't it be better to git grep on the entire code base to find all instances of the mis spelling and correct it that way further simplifying our code and doing away with the complexities of regex all together

**Comment by Umcaruje** [comment]
>Good evening ,

>Just out of curiosity wouldn't it be better to git grep on the entire code base to find all instances of the mis spelling and correct it that way further simplifying our code and doing away with the complexities of regex all together

Because we can't grep other people's projects.

**Comment by tresf** [comment]
> 1)  Despite the very low likelihood of someone making their own file or folder named "bassloopes", it still seems unsafe to change the path without knowing if this is an LMMS default or a user's personal file or folder.  Is this a concern?  What's a good way to go about preventing it?

* You can mitigate this by replacing `\\bassloopes\\` and `//bassloopes//`.
* You can mitigate this further by only running it as an upgrade task between applicable project versions.
* If you really (really) want this to be 100% foolproof, you can check for:
 *  `briff01.ogg`, `rave_bass01.ogg`, `rave_bass02.ogg`, `tb303_01.ogg`, `techno_bass01.ogg`, `techno_bass02.ogg`, `techno_synth01.ogg`, `techno_synth02.ogg`, `techno_synth03.ogg`, `techno_synth04.ogg`
* I do believe checking for each sample is overkill, btw.

> 2) Should I write my own upgrade for this?

Yes.

**Comment by T-boneG** [comment]
So I'm trying to make an upgrade right now and I'm coming across a few issues.

1) It seems like either I can change `VERSION_PATCH` in [https://github.com/LMMS/lmms/blob/master/CMakeLists.txt](url) or let `VERSION_SUFFIX` update itself when a new commit is made.  Which should it be?

2) If `VERSION_SUFFIX`, how do I know what that suffix will be for the upgrade if-statement?

3) Just in general while I've been messing around with the code, I can modify [https://github.com/LMMS/lmms/blob/master/CMakeLists.txt](url) and see changes when I build with any general change, however if I modify any of
```
SET(VERSION_MAJOR       "1")
SET(VERSION_MINOR       "1")
SET(VERSION_PATCH       "90")
```
the changes aren't reflected.  I have tried completely clearing my build and install directories to ensure no cache files are affecting this but the version will always be the same after running cmake even if, for example, I modify `SET(VERSION_PATCH       "90")` to be `SET(VERSION_PATCH       "91")`

**Comment by tresf** [comment]
@T-boneG thanks for looking into this and sorry for the delay in response.

To help prevent the distribution of incorrectly versioned binaries, we have a process which calculates the version based on the git commit.  Although it doesn't sound exactly like what's happening, you can use it to force a particular version which is different than the one in `CMakeLists.txt`.  Here's where the logic resides:

https://github.com/LMMS/lmms/blob/master/cmake/modules/VersionInfo.cmake#L25:L40

To use it, you can do the following:

```bash
cmake .. -DFORCE_VERSION=internal    # forces the version in CMakeLists.txt
cmake .. -DFORCE_VERSOIN=1.2.0       # force version to 1.2.0, etc.
```

By default the behavior is to use the version associated with the last commit, which may be undesirable when testing upgrade logic.

**Comment by T-boneG** [comment]
Hey this can be closed! :)
[https://github.com/LMMS/lmms/pull/2712](url)

**Comment by tresf** [comment]
@T-boneG, closing.  Also, if you put `closes #2617` in the body of your PR or commit id, that will do it as well. :+1: 

Thanks for the help!

**Comment by T-boneG** [comment]
Great, thanks for all the tips!  I look forward to staying involved!
